lib/shopping_cart.rb

```diff
@@ -0,0 +1,73 @@
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+    @full = false
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    cart_details = {
+      name: @name,
+      capacity: @capacity
+    }
+  end
+
+  def total_number_of_products
+    @products.sum do |product|
+      product.quantity
+    end
+  end
+
+  def is_full?
```

Comment:
We can simply this to `total_number_of_products > @capacity` and ditch the if else statment as this the code will always return a boolean value